pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java

```diff
@@ -1632,6 +1639,35 @@ public static Multimap<String, Metric> parseMetrics(String metrics) {
         return parsed;
     }
 
+    @Test
+    public void testRawMetricsProvider() throws IOException {
+        PrometheusMetricsProvider rawMetricsProvider = new PrometheusMetricsProvider();
+        try {
+            rawMetricsProvider.start(new PropertiesConfiguration());
+            rawMetricsProvider.getStatsLogger("test_raw")
+                .scopeLabel("topic", "persistent://public/default/test-v1")
+                .getOpStatsLogger("writeLatency")
+                .registerSuccessfulEvent(100, TimeUnit.NANOSECONDS);
+
+            getPulsar().addPrometheusRawMetricsProvider(rawMetricsProvider);
```

Comment:
Why are you doing that test?

The whole feature is about adding support of labels to BK client metrics. You changed `PrometheusMetricsProvider` to do that.
You changed the integration point of `PrometheusMetricsProvider` to `PrometheusMetricsGenerator`

I don't see any relation to `PrometheusRawMetricsProvider`.

I reiterate the last comment - you should test how suddenly BK client metrics have labels unless I'm missing something big here.

Reply:
The BK client metrics test is `testManagedLedgerBookieClientStats` and I updated the test to adjust for this change.
https://github.com/apache/pulsar/blob/90c554d3fa0efe0a24ca9af9946efa8908ccf2dc/pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java#L982-L1038

`testRawMetricsProvider` is mainly to test the `PrometheusRawMetricsProvider`. The relation is used by `getPulsar().addPrometheusRawMetricsProvider(rawMetricsProvider);` to register RawMetricsProvider to Pulsar service.